src/Aspire.Hosting/Backchannel/AppHostRpcTarget.cs

```diff
@@ -121,9 +121,22 @@ public async Task<DashboardUrlsState> GetDashboardUrlsAsync(CancellationToken ca
         // Wait for the dashboard to be healthy before returning the URL. This is to ensure that the
         // endpoint for the resource is available and the dashboard is ready to be used. This helps
         // avoid some issues with port forwarding in devcontainer/codespaces scenarios.
-        await resourceNotificationService.WaitForResourceHealthyAsync(
-            KnownResourceNames.AspireDashboard,
-            cancellationToken).ConfigureAwait(false);
+        try
+        {
+            await resourceNotificationService.WaitForResourceHealthyAsync(
+                KnownResourceNames.AspireDashboard,
+                WaitBehavior.StopOnResourceUnavailable,
+                cancellationToken).ConfigureAwait(false);
+        }
+        catch (DistributedApplicationException)
+        {
+            return new DashboardUrlsState
```

Comment:
Are we logging the underlying exception anywhere? Will it have any useful info?

Reply:
Fixing this up.